### DISCUSSION TITLE
Using an explicit MAC address for a codespace container vs local dev container

### DISCUSSION THREAD

**Discussion by worc4021** [community/community#43584] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hello!

I have been working with dev containers locally (in vscode) for a while now and have some useful ones set up that work for me locally. I've been working towards extending them to be accessible in the codespace environment and most of them are straightforward.

However, for a particular commercial software I have to use a license file that tries to validate the MAC address of the machine against a registered one. So in order to use it in a container, I need to launch the container with an explicit MAC address. 

So at the `docker run` level, I would be using `--mac-address=xyz` and hence in my local `devcontainer.json` I have it working with

```
"runArgs": [
        "--mac-address=${localEnv:LICENSE_MAC}"
]
```

When I try to run my **image** based dev container, I get the following error:

```
docker: Error response from daemon: conflicting options: mac-address and the network mode.
```

I was able to do all my other dev container to codespace migrations without having to _break_ the functionality for the counterpart (all dev containers work locally still and also in the codespaces). 

I would really like to be able to use this container in codespaces too, what would be the best-practise way to avoid having to set up additional networks and services etc to use a compose instead of a pure image?

Thanks!

**Comment by cwndrws** [comment, marked as answer]
Hi! 👋 Thanks for writing in!

You are unable to set the MAC address explicitly when using the default host networking mode in docker. In order to set the MAC address like this, you need to choose a different [docker networking mode](https://docs.docker.com/network/). Here's an example repo that I set up that works to set the MAC address explicitly for me: https://github.com/cwndrws/codespaces-networking-testing

The devcontainer settings:

```
{
    "runArgs": [
        "--network=bridge",
        "--mac-address=12:34:56:78:9a:bc"
    ]
}
```

Hope this helps!
